pkg/storage/unified/resource/search.go

```diff
@@ -0,0 +1,31 @@
+package resource
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// IndexableDocument can be written to a ResourceIndex
+type IndexableDocument interface {
+	// The ID must be unique across everything in the index
+	GetID() string
+}
+
+// Convert raw resource bytes into an IndexableDocument
+type DocumentBuilder interface {
+	// Convert raw bytes into an document that can be written
+	BuildDocument(ctx context.Context, key *ResourceKey, rv int64, value []byte) (IndexableDocument, error)
+}
+
+// Register how documents can be built for a resource
+type DocumentBuilderInfo struct {
```

Comment:
this is awkward sitting along like this 😬 , but it felt OK as i[nput parameters on SearchServer](https://github.com/grafana/grafana/compare/main...unified-search-exploration#diff-e4855c422ca8f979c4567e5d12f34ecd3a438a166dc8906ae3edda28bd5e4d20R114) rather than passing around provider providers 🤷🏻